app/controllers/lists_controller.rb

```diff
@@ -11,6 +11,9 @@ def index
   # GET /lists/1
   # GET /lists/1.json
   def show
+      @references = @list.references
+      .joins(:paper)
+      .order(('cached_votes_up DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
```

Comment:
will do.  Can I ask the logic?  is a function call like an SQL call?

On Thu, Sep 8, 2016 at 7:32 PM, Brian Kung <notifications@github.com> wrote:

> In app/controllers/lists_controller.rb
> <https://github.com/briankung/projectcredo/pull/70#discussion_r78113118>:
>
> > @@ -11,6 +11,9 @@ def index
> >    # GET /lists/1
> >    # GET /lists/1.json
> >    def show
> > +      @references = @list.references
> > +      .joins(:paper)
> > +      .order(('cached_votes_up DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
>
> Please never do this ('cached_votes_up DESC, ' if params[:sort] !=
> 'pub_date').to_s in the middle of a function call if you can avoid it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/70/files/e5240cef8e29cf5c4de4f689dc3026f25a5f6af5#r78113118>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZLCecUGHm5ALlCqGoZHpHD1FifoKks5qoKkcgaJpZM4J2i6I>
> .
>